### ISSUE TITLE
Send the firmware SYX, an error always occurs. 

### ISSUE THREAD

**Issue by Jpro600k** [image-et-son/p600fw#156]
When I connect the MIDI-OUT of the external keyboard to the MIDI-IN of the P600 internally and start the external keyboard, make a MIDI connection via the MIDI interface and send the firmware SYX, an error always occurs. When the power of the external keyboard was turned off, it was sent normally.
Is there any signal being output without pressing a key on the external keyboard? I don't understand the reason why SYX transmission from the outside cannot be received normally.
Perhaps it's a special combination that no one will do, but I'll send you a report just in case.
Connecting and merging external keyboards internally may be a problem.

**Comment by matrix12x** [comment]
I’ve had similar things happen on all gligli versions including my own.

**Comment by GrantMTG** [comment]
I don't completely follow how this connection is being made (keyboard plus PC to one MIDI input), but in any event it's possible that the keyboard is outputting Active Sensing and that the receiving firmware is not accounting for that?

**Comment by Jpro600k** [comment]
> I don't completely follow how this connection is being made (keyboard plus PC to one MIDI input), but in any event it's possible that the keyboard is outputting Active Sensing and that the receiving firmware is not accounting for that?

The cable is soldered to the contacts on the board of the MIDI OUT terminal on the main board of the MIDI keyboard, and from there, it is directly soldered to the MIDI terminal inside the P600 housing and directly connected.
Do not connect from the outside with a MIDI cable.
Simply connect the main board of the MIDI keyboard directly to the contacts inside the P600 chassis.
For velocity, it's a fully compatible P600 keyboard.
The transpose function cannot be used.
However, the arpeggiator function also does not work with LOCAL-ON. If you set it to LOCAL-OFF, you can use the arpeggiator. imogen taught me.

**Comment by matrix12x** [comment]
oh, I thought you meant that if you power on the P600 while holing does the buttons to put it in update mode, then connect the external MIDI interface, the P600 immediately shows an Error. Thats what mine does.
Or same thing if I have the external midi cable hooked up to the P600, but didn't plug in the USB end yet. When I plug the USB end, the same thing happens.

I'm assuming some thing is getting sent by my MIDI interface on power up.

**Comment by Jpro600k** [comment]
Although it has nothing to do with the problem, regarding the fitting of the bottom of the me board, I succeeded in lowering the keyboard surface to a problem-free height by cutting the plastic chassis at the bottom of the part where the 3372 chip hits in 6 places like the TM3 kit. I melted it with a soldering iron and cut it.

**Comment by Jpro600k** [comment]
When both MIDI IN and OUT are connected to a MIDI interface, CC Pedal was output from the MIDI OUT of P600. Does it have anything to do with the problem?

**Comment by image-et-son** [comment]
Hi there, I think that SysEx firmware upgrade is a delicate procedure and you should not expect it to work in any MIDI studio-like setup. On particular instability comes from the fact I the firmware consists of a sequence of complete SysEx commands (one for each memory storage page) but the software cannot handle/ignore MIDI commands that come in between the commands. 

I now see your issue with transpose by MIDI keyboard. I see if it can be changed so that MIDI transpose is possible in LOCAL OFF mode (like for the arpeggiator).

**Comment by image-et-son** [comment]
Hi Jpro600k, here's a trial version with MIDI transpose in local off mode supported. Please try it and tell me if it works. If yes, the change will then be incorporated in the alpha. Don't share this mod version as it will be difficult to keep track of what this version was for...
[p600firmware_beta2022_mod4Jpro600k_20220530.zip](https://github.com/image-et-son/p600fw/files/8798475/p600firmware_beta2022_mod4Jpro600k_20220530.zip)

**Comment by Jpro600k** [comment]
Hi, imogen
Thank you to the corresponding. I tested it.
When I turned LOCAL OFF and then pressed FROM TAPE and the keys on the external keyboard (Digital Piano), LOCAL ON and the TRANS value was displayed in the segment as if I pressed a key on the internal keyboard. 
However, it does not deviate from the pitch of the external keyboard. Even if you set it to -2, it seems to sound the same as 0. Normally, if you hear a sound on P600 with the set value, the sound should be out of sync with the external keyboard. It is not out of alignment.
As a result, the transpose setting could be obtained from a foreign key, but in P600 it was not actually transposed.
Am I doing something wrong?

**Comment by image-et-son** [comment]
> the TRANS value was displayed in the segment as if I pressed a key on the internal keyboard. However, it does not deviate from the pitch of the external keyboard. Even if you set it to -2, it seems to sound the same as 0. 

But I assume that the transposition you set via the MIDI keyboard works correctly for arpeggiator and sequencer, right?

**Comment by Jpro600k** [comment]
> But I assume that the transposition you set via the MIDI keyboard works correctly for arpeggiator and sequencer, right?

The arpeggiator and sequencer can be used without problems by operating from an external keyboard.

**Comment by Jpro600k** [comment]
> > the TRANS value was displayed in the segment as if I pressed a key on the internal keyboard. However, it does not deviate from the pitch of the external keyboard. Even if you set it to -2, it seems to sound the same as 0.
> 
> But I assume that the transposition you set via the MIDI keyboard works correctly for arpeggiator and sequencer, right?

When I recorded the sequencer using an external keyboard with P600 set to -6 in transpose, I was able to confirm that it was recorded at -6. However, the sound of a P600 played on an external keyboard is 0, not -6.

The value of the transpose updates the sequencer's key each time. I understand that the transpose is only inside.
After changing the key of the external keyboard, I recorded the P6 sequencer and confirmed that it was recorded at that pitch. If you want to record with a sequencer, transpose the external keyboard and it will record exactly to that key.
